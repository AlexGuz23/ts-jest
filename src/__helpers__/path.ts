import { resolve, relative } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function relativeToRoot(path: string) {
  return relative(ROOT, path)
}
