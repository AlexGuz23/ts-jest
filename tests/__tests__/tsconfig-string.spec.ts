jest.mock('path');

import getTSConfig from '../../dist/utils/get-ts-config';
import * as ts from 'typescript';
import * as path from 'path';
import cfg from '../__helpers__/jest-config';

describe('get ts config from string', () => {
  beforeEach(() => {
    // Set up some mocked out file info before each test
    (path as any).__setBaseDir('./tests/tsconfig-test');
  });

  describe('new behaviour (tsConfigFile & tsConfig)', () => {
    it('should correctly read my-tsconfig.json', () => {
      const result = getTSConfig(
        cfg.tsconfigTest({ tsConfigFile: 'my-tsconfig.json' }),
      );

      // snapshot would be enough here, but that adds a security in case we do not see it in a PR
      expect(result.target).toBe(ts.ScriptTarget.ES2015); // es6 resolves to ES2015 enum value
      expect(result).toMatchSnapshot();
    });

    it('should correctly resolve the "extends" directive', () => {
      const result = getTSConfig(
        cfg.tsconfigTest({ tsConfigFile: 'extends-tsconfig.json' }),
      );

      // snapshot would be enough here, but that adds a security in case we do not see it in a PR
      expect(result.target).toBe(ts.ScriptTarget.ES2015); // es6 resolves to ES2015 enum value
      expect(result).toMatchSnapshot();
    });

    it('should correctly override any config in the "extends" directive', () => {
      const result = getTSConfig(
        cfg.tsconfigTest({
          tsConfigFile: 'extends-with-overrides-tsconfig.json',
        }),
      );

      // snapshot would be enough here, but that adds a security in case we do not see it in a PR
      expect(result.target).toBe(ts.ScriptTarget.ES5);
      expect(result).toMatchSnapshot();
    });
  });
});
