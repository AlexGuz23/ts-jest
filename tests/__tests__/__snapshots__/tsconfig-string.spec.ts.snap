// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get ts config from string new behaviour (tsConfigFile & tsConfig) should correctly override any config in the "extends" directive 1`] = `
Object {
  "configFilePath": undefined,
  "declaration": false,
  "declarationMap": false,
  "emitDeclarationOnly": false,
  "inlineSourceMap": false,
  "inlineSources": true,
  "jsx": 2,
  "module": 1,
  "moduleResolution": 2,
  "noEmitOnError": true,
  "noImplicitAny": true,
  "sourceMap": true,
  "target": 1,
}
`;

exports[`get ts config from string new behaviour (tsConfigFile & tsConfig) should correctly read my-tsconfig.json 1`] = `
Object {
  "configFilePath": undefined,
  "declaration": false,
  "declarationMap": false,
  "emitDeclarationOnly": false,
  "inlineSourceMap": false,
  "inlineSources": true,
  "jsx": 2,
  "module": 1,
  "moduleResolution": 2,
  "noEmitOnError": true,
  "sourceMap": true,
  "target": 2,
}
`;

exports[`get ts config from string new behaviour (tsConfigFile & tsConfig) should correctly resolve the "extends" directive 1`] = `
Object {
  "configFilePath": undefined,
  "declaration": false,
  "declarationMap": false,
  "emitDeclarationOnly": false,
  "inlineSourceMap": false,
  "inlineSources": true,
  "jsx": 2,
  "module": 1,
  "moduleResolution": 2,
  "noEmitOnError": true,
  "sourceMap": true,
  "target": 2,
}
`;
